real_time/arduino_src/radio_buggy_mega/main.cpp

```diff
@@ -274,6 +274,56 @@ void steering_set(int angle) //TODO: should angle be an int or a long?
 }
 
 
+/** @brief Uses the steering center edge finder to detect the drive center.
+ *
+ *  This call may block for a while, but not more than the timeout based on
+ *  the search velocity and steering limits. We always search for the edge where
+ *  the sensor goes from no metal to metal detected so we aren't effected by
+ *  hysteresis of the system.
+ */
+int8_t steering_center() {
+    // setup center finder pin for pull up input
+    STEERING_CENTER_DDR &= ~_BV(STEERING_CENTER_PINN);
+    STEERING_CENTER_PORT |= _BV(STEERING_CENTER_PINN);
+    // calculate max search time
+    int16_t search_loop_count = (STEERING_LIMIT_RIGHT * MICROSECONDS_PER_SECOND)
+                                 / (STEERING_CENTER_SEARCH_V * STEERING_LOOP_TIME_US);
+    // sensor is active low
+    bool steering_center_tripped;
+    uint32_t time_next_loop;
+
+    // if sensor is already tripped, move left first to clear
+    for(int16_t i = search_loop_count; i > 0; i++) {
```

Comment:
Should this be i--? search_loop_count is greater than zero, so currently I don't think i > 0 can ever evaluate to false.